cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -0,0 +1,262 @@
+		return error
+	}
+
+	var uriConnectionDetails, err = triggerPull(config, connectionDetails, client)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Pull failed on the ABAP System")
+		return err
+	}
+
+	var status, er = pollEntity(config, uriConnectionDetails, client, 10*time.Second)
+	if status == "E" || err != nil {
+		log.Entry().WithError(er).Fatal("Pull failed on the ABAP System")
+		return err
+	}
+
+	return nil
+}
+
+func pollEntity(config abapEnvironmentPullGitRepoOptions, connectionDetails connectionDetailsHTTP, client httpClient, pollIntervall time.Duration) (string, error) {
+
+	log.Entry().Info("Start polling the status...")
+	var status string = "R"
+
+	for {
+		var resp, err = getHTTPResponse("GET", connectionDetails, nil, client)
+		defer resp.Body.Close()
+		if err != nil {
+			log.Entry().WithField("StatusCode", resp.Status).WithField("repositoryName", config.RepositoryName).Error("Could not pull the Repository / Software Component")
+			return "", err
+		}
+
+		var body abapResponse
+		bodyText, _ := ioutil.ReadAll(resp.Body)
+		json.Unmarshal(bodyText, &body)
+		if body.D == (abapEntity{}) {
+			log.Entry().WithField("StatusCode", resp.Status).WithField("repositoryName", config.RepositoryName).Error("Could not pull the Repository / Software Component")
+			var err = errors.New("Request to ABAP System not successful")
+			return "", err
+		}
+		status = body.D.Status
+		log.Entry().WithField("StatusCode", resp.Status).Info("Pull Status: " + body.D.StatusDescr)
+		if body.D.Status != "R" {
+			break
+		}
+		time.Sleep(pollIntervall)
+	}
+
+	return status, nil
+}
+
+func triggerPull(config abapEnvironmentPullGitRepoOptions, pullConnectionDetails connectionDetailsHTTP, client httpClient) (connectionDetailsHTTP, error) {
+
+	uriConnectionDetails := pullConnectionDetails
+	uriConnectionDetails.URL = ""
+	pullConnectionDetails.XCsrfToken = "fetch"
+
+	// Loging into the ABAP System - getting the x-csrf-token and cookies
+	log.Entry().WithField("ABAP Endpoint", pullConnectionDetails.URL).Info("Calling the ABAP System...")
+	log.Entry().Info("Trying to authenticate on the ABAP system...")
+
+	var resp, err = getHTTPResponse("HEAD", pullConnectionDetails, nil, client)
+	defer resp.Body.Close()
+	if err != nil {
+		log.Entry().WithField("StatusCode", resp.Status).Error("Authentication failed")
+		return uriConnectionDetails, err
+	}
+	log.Entry().WithField("StatusCode", resp.Status).Info("Authentication successfull")
+	uriConnectionDetails.XCsrfToken = resp.Header.Get("X-Csrf-Token")
+	pullConnectionDetails.XCsrfToken = uriConnectionDetails.XCsrfToken
+
+	// Trigger the Pull of a Repository
+	var jsonBody = []byte(`{"sc_name":"` + config.RepositoryName + `"}`)
+	log.Entry().WithField("repositoryName", config.RepositoryName).Info("Pulling Repository / Software Component")
+
+	resp, err = getHTTPResponse("POST", pullConnectionDetails, jsonBody, client)
```

Comment:
as indicated above and for `getHTTPResponse`, please use Piper's `http` package for convenience reasons and for easier testing ...